### PR TITLE
Added Study groups. Fixes #97

### LINKED ISSUE #97: /study
We need a section for our study groups. It should highlight the groups that are currently active as well as list the books we've done in the past (and have an area for us to write multi-person reviews of the books).

### PR DESCRIPTION
**Pull request by tlynam**
Based on Rails 4 upgrade pr
Added study groups
Can leave comments for study groups

### CODE REVIEW

File: app/controllers/groups_controller.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		@group = Group.find params[:id]
+	end		
+
+	private
+
+	def group_params
+		params.require(:group).permit(:book, :active, :github, :username)
+	end
+
+end
```
Comment (phiggins) [inline app/controllers/groups_controller.rb]: Your whitespace is off here too, like in the other PR. Seems like something you need to sort out in your editor. ;)

File: app/models/group.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Group < ActiveRecord::Base
+	has_many :comments
+	validates_presence_of :book
```
Comment (phiggins) [inline app/models/group.rb]: This probably isn't strictly necessary.

File: db/migrate/20150413224608_create_groups.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class CreateGroups < ActiveRecord::Migration
+  def change
+    create_table :groups do |t|
+      t.string :book, :null => false
+      t.boolean :active, :default => true
```
Comment (phiggins) [inline db/migrate/20150413224608_create_groups.rb]: This one should also be `:null => false`.

File: app/models/group.rb (hunk 2 above)
Comment (phiggins) [inline app/models/group.rb]: I was getting my wires crossed and I thought this was referring to a `Book` instance from the other PR. Maybe it should? I'm not sure if that's what we want for this though...

File: test/controllers/groups_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      post :create, group: {
+                             :book => "Winnie The Pooh",
+                             :active => false,
+                             :github => "www.yahoo.com"
+                            }
+    end
+
+    assert_redirected_to groups_path
+    get :index
+    assert_response :success
```
Comment (phiggins) [inline test/controllers/groups_controller_test.rb]: I made the same comment in the other PR, but I think there is something fancier you can do for redirects in tests.

**Comment by phiggins** [comment]
The code looks mostly good but I'll leave it for @zenspider's comments.

**Comment by zenspider** [comment]
Leaving for @kushali to review / approve.

**Pull request closed without merging**